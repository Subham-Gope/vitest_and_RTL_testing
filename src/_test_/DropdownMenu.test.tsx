import { describe, test, expect } from "vitest";
import { screen, render } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import DropDownMenu from "../components/DropdownMenu";

describe("Testing the DropdownMenu.sx file", () => {
  test("displays menu on hover", async () => {
    // setting the userEvent
    const user = userEvent.setup();

    // setting up the match snapshot
    const { container } = render(<DropDownMenu />);
    expect(container).toMatchSnapshot();

    const btn = screen.getByRole("button", { name: /help/i });
    expect(btn).toBeInTheDocument();

    // Initiating the user interactions
    await user.hover(btn);

    const menu = screen.getByRole("menu", { name: /menu/i });
    expect(menu).toBeInTheDocument();

    await user.unhover(btn);
    expect(menu).not.toBeInTheDocument();
  });
});
