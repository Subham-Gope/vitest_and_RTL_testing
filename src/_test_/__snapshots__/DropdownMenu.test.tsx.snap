// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Testing the DropdownMenu.sx file > displays menu on hover 1`] = `
<div>
  <div>
    <div
      aria-label="help"
      role="button"
    >
      Menu
    </div>
  </div>
</div>
`;
